later/lib/widgets/buttons/primary_button.dart

```diff
@@ -12,7 +13,16 @@ class PrimaryButton extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final LaterPalette laterPalette =
+        Theme.of(context).extension<LaterPalette>()!;
+    final LaterTypography laterTypography =
+        Theme.of(context).extension<LaterTypography>()!;
     return TextButton(
+      style: TextButton.styleFrom(
+        foregroundColor: laterPalette.primaryColor,
+        disabledForegroundColor: laterPalette.primaryDisabledColor,
+        textStyle: laterTypography.text1,
+      ),
```

Comment:
Prefer "dividing" the code into logical blocks. Here we have:
- Defining two theme variables
- Returning a widget

Between these two we should add an empty line to visually divide these two blocks. It increases readability. In the future, with larger code blocks, it will be even nicer.

```suggestion
    final LaterPalette laterPalette =
         Theme.of(context).extension<LaterPalette>()!;
     final LaterTypography laterTypography =
         Theme.of(context).extension<LaterTypography>()!;

     return TextButton(
       style: TextButton.styleFrom(
         foregroundColor: laterPalette.primaryColor,
         disabledForegroundColor: laterPalette.primaryDisabledColor,
         textStyle: laterTypography.text1,
       ),
```